bundles/org.openhab.binding.homekit/src/main/java/org/openhab/binding/homekit/internal/handler/HomekitBaseAccessoryHandler.java

```diff
@@ -0,0 +1,874 @@
+        Instant timeout = Instant.now().plus(HANDLER_INITIALIZATION_TIMEOUT);
+        while (!bridgedThingsInitialized() && Instant.now().isBefore(timeout)) {
+            try {
+                Thread.sleep(100);
+            } catch (InterruptedException e) {
+                Thread.currentThread().interrupt(); // shutting down; restore interrupt flag, and exit immediately
+                return;
+            }
+        }
+        onConnectedThingAccessoriesLoaded();
+        onThingOnline();
+    }
+
+    /**
+     * Returns the accessory ID. For bridges and accessories this is always 1. Whereas for
+     * bridged accessories it comes from the thing's configuration parameter value.
+     *
+     * @return the accessory ID, or null if it cannot be determined
+     */
+    protected @Nullable Long getAccessoryId() {
+        if (isBridgedAccessory) {
+            if (getConfig().get(CONFIG_ACCESSORY_ID) instanceof BigDecimal accessoryId) {
+                try {
+                    return accessoryId.longValue();
+                } catch (NumberFormatException e) {
+                }
+            }
+            logger.debug("{} missing or invalid accessory id", thing.getUID());
+            return null;
+        }
+        return 1L;
+    }
+
+    @Override
+    public void handleRemoval() {
+        cancelRefreshTasks();
+        if (isBridgedAccessory) {
+            updateStatus(ThingStatus.REMOVED);
+        } else {
+            scheduler.submit(() -> {
+                if (unpairInner().startsWith(ACTION_RESULT_OK)) {
+                    updateStatus(ThingStatus.REMOVED);
+                }
+            });
+        }
+    }
+
+    @Override
+    public void initialize() {
+        eventedCharacteristics.clear();
+        accessories.clear();
+        isBridgedAccessory = getBridge() instanceof Bridge;
+        if (!isBridgedAccessory) {
+            scheduleConnectionAttempt();
+        }
+        updateStatus(ThingStatus.UNKNOWN);
+    }
+
+    /**
+     * Restores an existing pairing.
+     * Updates the thing status accordingly.
+     */
+    private synchronized boolean verifyPairing() {
+        isConfigured = false;
+        Long accessoryId = checkedAccessoryId();
+        String ipAddress = checkedIpAddress();
+        String macAddress = checkedMacAddress();
+        String hostName = checkedHostName();
+        if (accessoryId == null || ipAddress == null || macAddress == null || hostName == null) {
+            return false; // configuration error
+        }
+        isConfigured = true;
+
+        // check if we have a stored key
+        Ed25519PublicKeyParameters accessoryKey = keyStore.getAccessoryKey(macAddress);
+        if (accessoryKey == null) {
+            logger.debug("{} no stored pairing credentials", thing.getUID());
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    i18nProvider.getText(bundle, "error.not-paired", "Not paired", null));
```

Comment:
Here and other places where `updateStatus` is written in a similar way - why not simply:
```suggestion
            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
                    "@text/error.not-paired");
```

?

That would remove the need for double maintenance of the texts.

Reply:
see above